pkg/oci/oci.go

```diff
@@ -372,31 +387,29 @@ func DeleteGadgetImage(ctx context.Context, image string) error {
 	return deleteTree(ctx, ociStore, descriptor)
 }
 
-func getTagFromImage(image string) (string, error) {
-	repo, err := reference.Parse(image)
-	if err != nil {
-		return "", fmt.Errorf("parsing image %q: %w", image, err)
-	}
-	tagged, ok := repo.(reference.Tagged)
-	if !ok {
-		return "latest", nil
-	}
-	return tagged.Tag(), nil
-}
-
-func getRepositoryFromImage(image string) (string, error) {
-	repo, err := reference.Parse(image)
-	if err != nil {
-		return "", fmt.Errorf("parsing image %q: %w", image, err)
+// splitIGDomain splits a repository name to domain and remote-name.
+// If no valid domain is found, the default domain is used. Repository name
+// needs to be already validated before.
+func splitIGDomain(name string) (domain, remainder string) {
+	i := strings.IndexRune(name, '/')
+	if i == -1 || (!strings.ContainsAny(name[:i], ".:") && name[:i] != localhost && strings.ToLower(name[:i]) == name[:i]) {
```

Comment:
Just for my understanding, for `strings.ToLower(name[:i]) == name[:i])` we assume it it a domain name since it contains capital letters?

Reply:
>  domains are supported to be lower case, aren't they?

IIUC, domains are case insensitive, https://serverfault.com/a/261344

> It was introduced in distribution/reference@ ca2a562, the tests cases added parse the domain with upper case, not sure what this is for and the commit doesn't contain anymore
information.

I found more information here, https://github.com/distribution/distribution/pull/2979 and it gives more context why they did it.